.test.sh

```diff
@@ -2,5 +2,5 @@
 set -euo pipefail
 
 python -m mypy ./src
-python -m pyright .
+python -m pyright ./src
```

Comment:
do we need both `mypy` and `pyright`? And I also suggest to not limit it to `src` directory only, but also include `tests` directory, to make sure we use correct types on testing.